sql/core/src/test/scala/org/apache/spark/sql/sources/CreateTableAsSelectSuite.scala

```diff
@@ -235,10 +235,10 @@ class CreateTableAsSelectSuite extends DataSourceTest with SharedSparkSession {
   }
 
   test("create table using as select - with overridden max number of buckets") {
-    def createTableSql(numBuckets: Int): String =
+    def createTableSql(tablePath: String, numBuckets: Int): String =
       s"""
          |CREATE TABLE t USING PARQUET
-         |OPTIONS (PATH '${path.toURI}')
+         |OPTIONS (PATH '$tablePath')
```

Comment:
is this change necessary? it `beforeEach` we create a new temp dir and assign it to `path`.

Reply:
Yes, in this test case, the function  `createTableSql ` will be called twice inside `Seq(100001, maxNrBuckets).foreach(...)`. As this PR doesn't allow CTAS with a non-empty location, the second call of  `createTableSql` will fail if the same `path` used. That is why for each iteration a new table location is created